src/main/java/com/snowflake/kafka/connector/SnowflakeSinkConnector.java

```diff
@@ -156,6 +176,12 @@ public Class<? extends Task> taskClass() {
    */
   @Override
   public List<Map<String, String>> taskConfigs(final int maxTasks) {
+    // create all necessary clients, evenly mapping tasks to clients
+    if (this.usesStreamingIngestion) {
+      IngestSdkProvider.streamingClientManager.createAllStreamingClients(
```

Comment:
would prefer to do this in start(), but couldnt figure out how to get the maxtasks / pull from the config there?

Reply:
If doing it in `start` isn't feasible due to the config not being initialized then I'd just add a comment here saying why this is here.